setup.py

```diff
@@ -734,27 +690,34 @@ def get_git_version_suffix():
         return get_git_commit_hash()
 
 
+# keep it separate for easy substitution
+TRITON_VERSION = "3.3.0" + get_git_version_suffix() + os.environ.get("TRITON_WHEEL_VERSION_SUFFIX", "")
+
+package_dirs = dict(get_package_dirs())
+extra_packages = [x for x in package_dirs if x != ""]
+
 setup(
     name=os.environ.get("TRITON_WHEEL_NAME", "triton"),
-    version="3.3.0" + get_git_version_suffix() + os.environ.get("TRITON_WHEEL_VERSION_SUFFIX", ""),
+    version=TRITON_VERSION,
     author="Philippe Tillet",
     author_email="phil@openai.com",
     description="A language and compiler for custom Deep Learning operations",
     long_description="",
-    install_requires=["setuptools>=40.8.0"],
-    packages=get_packages(),
+    install_requires=[
+        "setuptools>=40.8.0",
+        "importlib-metadata; python_version < '3.10'",
```

Comment:
What is the reason to introduce new `importlib-metadata` dependency ?

Reply:
`importlib.metadata` is used by the new backend finder. Unfortunately, the API changed in Python 3.10, and the usual way of dealing with that is to pull `importlib-metadata` backport for older Pythons.